flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/SubtaskCheckpointCoordinatorImpl.java

```diff
@@ -183,6 +235,42 @@ public void notifyCheckpointComplete(long checkpointId, OperatorChain<?, ?> oper
 		env.getTaskStateManager().notifyCheckpointComplete(checkpointId);
 	}
 
+	@Override
+	public void notifyCheckpointAborted(long checkpointId, OperatorChain<?, ?> operatorChain, Supplier<Boolean> isRunning) throws Exception {
+
+		if (isRunning.get()) {
+			LOG.debug("Notification of aborted checkpoint for task {}", taskName);
+			// only happens when the task always received checkpoints to abort but never trigger or executing.
+			if (abortedCheckpointIds.size() >= maxRecordAbortedCheckpoints) {
+				abortedCheckpointIds.pollFirst();
+			}
+
+			channelStateWriter.abort(checkpointId, new NotifiedCheckpointAbortedException(checkpointId));
```

Comment:
`try/catch`?
same for the operators

Reply:
Would you please give more clear explanations for this? Did you mean we cannot call `channelStateWriter.abort` twice. Moreover, what do you mean for `same for the operators`?

Reply:
Sorry for not being clear. We shouldn't call `channelStateWriter.abort` before receiving a cancelation marker.

> Moreover, what do you mean for same for the operators?

Added a separate comment.

Reply:
If the async checkpoint on task side is aborted via RPC message, and we did not call `channelStateWriter.abort`. If above stream did not broadcast a cancellation marker to downstream, what happened to `channelStateWriter` as that is not called as aborted.

Reply:
In that case, `CheckpointBarrierHandler` will call abort on the next barrier.
Though I don't think it's possible that task skips the cancellation marker.